src/main/java/training/chessington/model/Board.java

```diff
@@ -53,27 +54,44 @@ public void placePiece(Coordinates coords, Piece piece) {
     }
 
     public boolean moveIsBlocked(Move move) {
-        
-        int fromRow = move.getFrom().getRow();
-        int fromCol = move.getFrom().getCol();
-        int toRow = move.getTo().getRow();
-        int toCol = move.getTo().getCol();
-        
-        boolean isHorizontal = fromRow == toRow;
-        boolean isVertical = fromCol == toCol;
-        boolean isDiagonal = (fromRow - toRow) == (fromCol - toCol);
-        
-        if (isHorizontal) {
-            return linearMoveIsBlocked(fromRow, fromCol, toCol, false);
-        } else if (isVertical) {
-            return linearMoveIsBlocked(fromCol, fromRow, toRow, true);
+        Function<Coordinates, Coordinates> directionalStep = this.calculateDirectionalStep(move.getFrom(), move.getTo());
+        return this.linearMoveIsBlocked(move, directionalStep);
+    }
+
+    private Function<Coordinates, Coordinates> calculateDirectionalStep(Coordinates start, Coordinates end) {
+        if (start.equals(end)) {
+            return (position) -> position;
+        } else if (start.getRow() == end.getRow()) {
+            int colDisplacement = end.getCol() - start.getCol();
+            int direction = colDisplacement / Math.abs(colDisplacement);
+            return (position) -> position.plus(0, direction);
+        } else if (start.getCol() == end.getCol()) {
+            int rowDisplacement =  end.getRow() - start.getRow();
+            int direction = rowDisplacement / Math.abs(rowDisplacement);
+            return (position) -> position.plus(direction, 0);
+        } else {
+            int rowDisplacement = end.getRow() - start.getRow();
+            int colDisplacement = end.getCol() - start.getCol();
+            int rowDirection = rowDisplacement / Math.abs(rowDisplacement);
+            int colDirection = colDisplacement / Math.abs(colDisplacement);
+            return (position) -> position.plus(rowDirection, colDirection);
+        }
+    }
+
+    private boolean linearMoveIsBlocked(Move move, Function<Coordinates, Coordinates> directionalStep) {
+        Coordinates endSquare = move.getTo();
+        Coordinates nextSquare = directionalStep.apply(move.getFrom());
+        while (!nextSquare.equals(endSquare) && this.contains(nextSquare)) {
+            if (this.get(nextSquare) != null) {
+                return true;
+            }
+            nextSquare = directionalStep.apply(nextSquare);
         }
-        // else if (isDiagonal) {...}
         return false;
     }
 
-    private boolean linearMoveIsBlocked(int lineIndex, int fromIndex, int toIndex, boolean moveIsVertical) {
-        for (int spaceIndex = Math.min(fromIndex + 1, toIndex); spaceIndex <= Math.max(toIndex, fromIndex - 1); spaceIndex++) {
+    private boolean lateralMoveIsBlocked(int lineIndex, int fromIndex, int toIndex, boolean moveIsVertical) {

```

Comment:
This function might be superseded by the one above? I can't think of when this would be needed over the other.